orchagent/copporch.cpp

```diff
@@ -0,0 +1,479 @@
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pir:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_action_green_field)
+            {
+                SWSS_LOG_DEBUG("green action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_red_field) 
+            {
+                SWSS_LOG_DEBUG("red action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_yellow_field)
+            {
+                SWSS_LOG_DEBUG("yellowaction:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown copp field specified:%s\n", fvField(*i).c_str());
+                return task_process_status::task_invalid_entry;
+            }
+        }
+
+        if (m_trap_map.find(key) != m_trap_map.end())
+        {
+            SWSS_LOG_DEBUG("found existing trap_group object:%s", key.c_str());
+            if(!policer_attribs.empty())
+            {
+                attr.id = SAI_HOSTIF_TRAP_GROUP_ATTR_POLICER;
+                attr.value.oid = SAI_NULL_OBJECT_ID;
+                if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->get_trap_group_attribute(m_trap_map[key], 1, &attr)))
+                {
+                    SWSS_LOG_ERROR("Failed to get policer for trap gorup:%s, error:%d\n", key.c_str(), sai_status);
+                    return task_process_status::task_failed;
+                }
+                SWSS_LOG_DEBUG("Obtained policer object:%llx for trap_group:%s", attr.value.oid, key.c_str());
+                policer_id = attr.value.oid;                
+                SWSS_LOG_DEBUG("Applying settings to existing policer:%llx for trap_group:%llx, name:%s", policer_id, m_trap_map[key], key.c_str());
+                if(SAI_NULL_OBJECT_ID == policer_id)
+                {
+                    SWSS_LOG_WARN("Creating policer for existing Trap group:%llx (name:%s).", m_trap_map[key], key.c_str());
+                    if (SAI_STATUS_SUCCESS != (sai_status = sai_policer_api->create_policer(&policer_id, policer_attribs.size(), policer_attribs.data())))
```

Comment:
>policer_id [](start = 93, length = 10)

once you have created the policer_id, I do not see how you assign the policer_id to the trap_group.